src/Compilers/CSharp/Portable/Binder/Semantics/INonNullTypesContext.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// Instead of storing a bool to tell us if we're in a NonNullTypes context,
+    /// we use this interface to pull on that information lazily.
+    /// </summary>
+    internal interface INonNullTypesContext
+    {
+        bool NonNullTypes { get; }
+    }
+
+    internal sealed class NonNullTypesContextTrue : INonNullTypesContext
```

Comment:
Nit: Rather than True / False suffix consider `EnableNonNullTypes` / `DisableNonNullTypes`.

Reply:
I was actually thinking to change to `NonNullTypesTrueContext`, since that's what we tend to speak out loud ("[NonNullTypes(true)] context"). We don't typically say "enabled" or "disabled", although it's correct.
Sounds alright?

Reply:
That sounds fine to me.